libexec/query.ts

```diff
@@ -40,7 +40,7 @@ if (versions) {
   Deno.exit(0)
 }
 
-const version = pkg.constraint.single() ?? panic()
+const {version} = await usePantry().resolve(pkg) ?? panic()
```

Comment:
yeah this changes the behavior and is thus not acceptable.

Reply:
Actually, this change was suggested by @jhheider as a hack to preserve `version.raw` for single constraints. (Some like zlib were breaking because version.raw included trailing .0, which they don't expect.

Reply:
We can also probably do this:
```ts
const pkg_version = pkg.constraint.single() ?? panic()
const pantry_ver = await usePantry().resolve(pkg) ?? panic()
const version = (pkg_version.raw == pantry_ver.version.raw) ? pantry_ver.version : pkg_version
```
It'll have the same behaviour for cases where the .raw matches (for normal cases). For others, it'll change that to fix it.

Reply:
@mxcl Any suggestions?